packages/chart/src/components/LinearChart.tsx

```diff
@@ -308,13 +323,33 @@ const LinearChart: React.FC<LinearChartProps> = props => {
   }, [dataRef?.isIntersecting, config.animate])
 
   useEffect(() => {
-    const textElement = document.querySelector(`#suffix`)
-    if (!textElement && !suffixWidth) return
-    if (!textElement) return setSuffixWidth(0)
-    const textWidth = textElement.getBBox().width
-    setSuffixWidth(textWidth)
+    const suffixEl = suffixRef.current
+    if (!suffixEl && !suffixWidth) return
+    if (!suffixEl) return setSuffixWidth(0)
+    const suffixElWidth = suffixEl.getBBox().width
+    setSuffixWidth(suffixElWidth)
   }, [config.dataFormat.suffix, config.dataFormat.onlyShowTopPrefixSuffix])
 
+  // forest plot x-axis label positioning
+  useEffect(() => {
```

Comment:
https://react.dev/learn/you-might-not-need-an-effect

Reply:
This was a great read! Definitely converted me to team minimize useEffects.  I think I still need these useEffects here when doing calculations based on the size of rendered elements.  The size of an element isn't available until after it has been rendered to the DOM so unfortunately I think these ones need to stay for now.

Reply:
Why use a state to hold the calculations? the rest of the component doesn't need to know about that, only the element on the page. I'd recommend using a ref and doing a direct DOM manipulation in that case. That way you don't rerender the whole page when you setForestXLabelY.

I did something similar to that here:
https://github.com/CDCgov/cdc-open-viz/pull/1581/files#diff-002175af9bf20550dd9e6e59cfbbf6f86f1986300d62c44289b1aac9ae703c57

Reply:
Oh nice! Ok, refactored.